### ISSUE TITLE
Blank Terminal Screen systemdspawner

### ISSUE THREAD

**Issue by jakepaw** [jupyterhub/jupyterhub#2738]
Hello everyone,

I have been working with jupyterhub using the systemd spawner. I am able to connect to the hub spawn a process and see files in my home dir, yet when I want to start a terminal i am left with a blank screen. I tried setting my SHELL=bash, and keepo the env variables from the user logged in. The passwd file states the shell as well. Any insight would be appreciated. Here is what I have for systemdspawner configs:

c.SystemdSpawner.default_shell = '/bin/bash'
c.SystemdSpawner.mem_limit = '4G'
c.SystemdSpawner.cpu_limit = 4.0
c.SystemdSpawner.workingdir = '/home/{USERNAME}'
c.SystemdSpawner.extra_paths = ['/home/{USERNAME}/conda/bin']
c.SystemdSpawner.unit_name_template = 'jupyter-{USERNAME}-singleuser'
c.SystemdSpawner.environment_variables = ['SHELL=/bin/bash']

I feel like I am missing a simple config to have this run....

Other configs may be related / attempted to set:
c.NotebookApp.terminals_enabledBool = True
c.NotebookApp.terminado_settings={'shell_command': ['/bin/bash']}
c.SystemdSpawner.environment = {'SHELL': ['/bin/bash']}

![blank_terminal_jupyterhub](https://user-images.githubusercontent.com/13523932/65333745-11fb5100-db8f-11e9-954e-2e94b06a39f3.PNG)

Thanks for any insight :)

**Comment by ryogesh** [comment]
1) What are the versions of OS, Jupyterhub, Spawner...
2) Enable Debug on the hub, spawner, notebook... and check the log msgs
3) Launch jupyterlab /lab and check behavior of the terminal launch from lab

**Comment by jakepaw** [comment]
1. OS (RHEL 7.6), Jupyterhub (1.0.0), SystemdSpawner (0.13)
2. Enabled Debug on hub, spawner, and notebook


[terminal_log.txt](https://github.com/jupyterhub/jupyterhub/files/3652853/terminal_log.txt)

**Comment by willingc** [comment]
Hi @jakepaw Thanks for opening this issue. Are you able to launch notebooks properly?

**Comment by ryogesh** [comment]
Can you check on the systemd version?

https://github.com/jupyterhub/systemdspawner#requirements

**Comment by willingc** [comment]
Thanks @ryogesh. I was just about to suggest the same. [Centos 7 has a version of systemd that is too old for support.](https://github.com/jupyterhub/systemdspawner#centos-7)

**Comment by willingc** [comment]
> c.SystemdSpawner.environment_variables = ['SHELL=/bin/bash']

I believe this can be left unset since you have set `c.SystemdSpawner.default_shell = '/bin/bash'` since this configurable trait should set SHELL.

> c.NotebookApp.terminals_enabledBool = True
> c.NotebookApp.terminado_settings={'shell_command': ['/bin/bash']}
> c.SystemdSpawner.environment = {'SHELL': ['/bin/bash']}

Perhaps unset these as well.

**Comment by jakepaw** [comment]
@willingc and @ryogesh
Since opening I have taken away the env variable It was set incorrectly, and changed to a dictionary. I am running systemd 219

Note: I also replaced the file (systemdspawner.systemd.py) with this file https://raw.githubusercontent.com/jupyterhub/systemdspawner/master/systemdspawner/systemd.py because of an issue i was having using systemd 219, this allowed me to view my Notebook page.

**Comment by willingc** [comment]
@jakepaw Is it all working now?

**Comment by jakepaw** [comment]
@willingc Not yet, still unsure of what the is going on with the terminal. Below are my configs which I should have originally posted, as well as my conda -list output just for reference. 

[Terminal_error_configs.txt](https://github.com/jupyterhub/jupyterhub/files/3657602/Terminal_error_configs.txt)

[packages.txt](https://github.com/jupyterhub/jupyterhub/files/3657608/packages.txt)

**Comment by willingc** [comment]
In `jupyter_config.py`, do you have these set? [ref](https://jupyter-notebook.readthedocs.io/en/stable/config.html)
`NotebookApp.terminado_settingsDict`
`NotebookApp.terminals_enabledBool`

Try upgrading the `terminado` package to a more recent version. Current is 0.8.2 and it looks like you have 0.6.0 installed.

**Comment by jakepaw** [comment]
@willingc 

I have set the following variables:
'''
c.NotebookApp.terminals_enabledBool = True
c.NotebookApp.terminado_settings= {'shell_command': ['/bin/bash']}
'''

**Comment by willingc** [comment]
@jakepaw Before we dig into the proxy, I want to verify that the notebooks work presently. It is only the terminal that does not work. Is that correct?

If so, please post the ConfigProxy debug output like above for a successful notebook launch. Thanks.

**Comment by jakepaw** [comment]
@willingc 
Here is the debug output of the successful startup that happens each time. I am able to log into jupyterhub and see my files (see attached screengrab below). Thank you!

![Capture](https://user-images.githubusercontent.com/13523932/65732137-75591780-e096-11e9-9e0d-c32496d99f2a.PNG)

[debug_log.txt](https://github.com/jupyterhub/jupyterhub/files/3660291/debug_log.txt)

**Comment by ryogesh** [comment]
1) In the log file i see the line user_id admin. Can you launch a terminal session as admin (say using putty)?
2) Disable suystemspawner and check if you can run the terminal with localspawner?
3) Is SELinux enabled?, can you check the audit logs.

**Comment by willingc** [comment]
@minrk Do you have any thoughts on what is missing/incorrect in the configuration that would result in a black terminal but access to the file browser? Using LDAPAuthenticator and SystemdSpawner. The prior message has the debug output and below is the current config related to URLs/IPs [full config is .

```
c.JupyterHub.proxy_class = 'jupyterhub.proxy.ConfigurableHTTPProxy'
c.ConfigurableHTTPProxy.api_url = 'my_ip:8081'
c.ConfigurableHTTPProxy.should_start = True
c.ConfigurableHTTPProxy.auth_token = "CONFIGPROXY_AUTH_TOKEN" 
c.SystemdSpawner.unit_name_template = 'jupyter-{username}-singleuser'
c.SystemdSpawner.debug = True
c.JupyterHub.spawner_class = 'systemdspawner.SystemdSpawner'
c.JupyterHub.hub_connect_ip = 'my_ip'
c.JupyterHub.bind_url = 'my_hostname:8002'
c.JupyterHub.authenticator_class = 'ldapauthenticator.LDAPAuthenticator'
c.LDAPAuthenticator.server_address = 'ldap_connect'
c.LDAPAuthenticator.server_port = 636
```

**Comment by jakepaw** [comment]
@willingc
Thank you everyone for the help. Working through it this morning I followed the steps of @ryogesh and used local spawner. From here I saw that there was an error 
```
AttributeError: 'NoneType' object has no attribute 'start' 
```
That stemmed from zmq and I found this https://github.com/jupyter/notebook/issues/4441
This allowed me to resolve the issue by upgrading to notebook 5.7.5 with local spawner..

From here I was able to re-install jupyterhub-systemdspawner and change the systemdspawner/systemd.py to the one I referenced (https://raw.githubusercontent.com/jupyterhub/systemdspawner/master/systemdspawner/systemd.py)

After setting all my params for systemd.spawner I was able to successfully open the terminal! 

TLDR: upgraded notebook from 5.0.0 to 5.7.5
